app/models/constants.ts

```diff
@@ -32,3 +32,114 @@ export const NATIONAL_AVERAGES: nationalAverageType = {
 };
 
 export const MAINTENANCE_LEVELS = [0.015, 0.02, 0.0375] as const;
+
+/** Type for storing component values and depreciation*/
+export type componentBreakdownType = {
+  /** Component value as percentage of total house value */
+  percentageOfHouse: number, 
+  /** Percentage of the component's total value that is written-down yearly */
+  depreciationPercentageYearly: number,
+  /** Percentage of the yearly maintenance spend allocated to component */
+  percentOfMaintenanceYearly: number
+}
+
+export type houseBreakdownType = {
```

Comment:
It might be simpler to make a union of the keys, and then define this as as record.

```ts
type Component = "foundations" | "structureEnvelope" | ... 
type HouseBreakdown = Record<Component, ComponentBreakdown>
```